lib/sisimai/bite/email/gsuite.rb

```diff
@@ -0,0 +1,273 @@
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+
+            if connvalues == connheader.keys.size
+              # Final-Recipient: rfc822; kijitora@example.de
+              # Action: failed
+              # Status: 5.0.0
+              # Remote-MTA: dns; 192.0.2.222 (192.0.2.222, the server for the domain.)
+              # Diagnostic-Code: smtp; 550 #5.1.0 Address rejected.
+              # Last-Attempt-Date: Fri, 24 Mar 2017 23:34:10 -0700 (PDT)
+              v = dscontents[-1]
+
+              if cv = e.match(/\A[Ff]inal-[Rr]ecipient:[ ]*(?:RFC|rfc)822;[ ]*(.+)\z/)
+                # Final-Recipient: rfc822; kijitora@example.de
+                if v['recipient']
+                  # There are multiple recipient addresses in the message body.
+                  dscontents << Sisimai::Bite.DELIVERYSTATUS
+                  v = dscontents[-1]
+                end
+                v['recipient'] = cv[1]
+                recipients += 1
+
+              elsif cv = e.match(/\A[Aa]ction:[ ]*(.+)\z/)
+                # Action: failed
+                v['action'] = cv[1].downcase
+
+              elsif cv = e.match(/\A[Ss]tatus:[ ]*(\d[.]\d+[.]\d+)/)
+                # Status: 5.0.0
+                v['status'] = cv[1]
+
+              elsif cv = e.match(/\A[Rr]emote-MTA:[ ]*(?:DNS|dns);[ ]*(.+)\z/)
+                # Remote-MTA: dns; 192.0.2.222 (192.0.2.222, the server for the domain.)
+                v['rhost'] = cv[1].downcase
+                v['rhost'] = '' if v['rhost'] =~ /\A\s+\z/  # Remote-MTA: DNS;
+
+              elsif cv = e.match(/\A[Ll]ast-[Aa]ttempt-[Dd]ate:[ ]*(.+)\z/)
+                # Last-Attempt-Date: Fri, 24 Mar 2017 23:34:10 -0700 (PDT)
+                v['date'] = cv[1]
+
+              else
+                if cv = e.match(/\A[Dd]iagnostic-[Cc]ode:[ ]*(.+?);[ ]*(.+)\z/)
+                  # Diagnostic-Code: smtp; 550 #5.1.0 Address rejected.
+                  v['spec'] = cv[1].upcase
+                  v['diagnosis'] = cv[2]
+                else
+                  # Append error messages continued from the previous line
+                  if endoferror && ( v['diagnosis'] && v['diagnosis'].size > 0 )
+                    endoferror = true if e.empty?
+                    endoferror = true if e =~ /\A--/
+
+                    next if endoferror
+                    next unless e =~ /\A[ ]/
+                    v['diagnosis'] += e
+                  end
+                end
+              end
+
+            else
```

Comment:
__[rubocop]__
* Performance/StartWith: Use `String#start_with?` instead of a regex match anchored to the beginning of the string. [:ref](https://github.com/JuanitoFatas/fast-ruby#stringmatch-vs-stringstart_withstringend_with-code-start-code-end)

<!-- Comment by [SideCI](https://sideci.com) -->